### PR TITLE
Add Kfdef for Data Science pipelines operator

### PR DESCRIPTION
**Pull request by rimolive**
Enables Data Science Pipelines Operator as a Kfdef CR

## Description
This will add Data Science Pipelines Operator as a Kfdef to deploy DSP in a multi-stack architecture. DSPO will be deployed on `redhat-ods-applications` namespace.

## Merge criteria:
PRs for [RHODS-6926](https://issues.redhat.com/browse/RHODS-6926), [RHODS-739](https://issues.redhat.com/browse/RHODS-7399), and [RHODS-7400](https://issues.redhat.com/browse/RHODS-7400) must be merged first

- [X] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] For commits that came from upstream, `[UPSTREAM]` has been prepended to the commit message.
- [X] JIRA link(s): https://issues.redhat.com/browse/RHODS-7151
- [ ] The Jira story is acked.
- [ ] Live build image: 
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work.
- [ ] QE contact acknowledges that this has been tested and is approved for merge.

### CODE REVIEW

File: deploy.sh [hunk 1]
```diff
@@ -12,7 +12,7 @@
 # distributed under the License is distributed on an "AS IS" BASIS,
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
-# limitations under the License.
+# limitations under the License./
```
Comment (HumairAK) [inline deploy.sh:15]: typo?

**Comment by VaishnaviHire** [comment]
If we are introducing deployments in `redhat-ods-applications`, we would need to update NetworkPolicy for that namespace https://github.com/red-hat-data-services/odh-deployer/blob/main/network/applications_network_policy.yaml

**Comment by rimolive** [comment]
Mental note: Don't forget to add the parameters for all DSP images.

**Comment by HumairAK** [comment]
@VaishnaviHire, there is no ingress/egress traffic for this application except for internal traffic to the api server, this is an operator that submits resources to the cluster. It has the appropriate rbac permission. In this case, it seems to me the network policy doesn't need to be updated -- can you confirm?

**Comment by VaishnaviHire** [comment]
> @VaishnaviHire, there is no ingress/egress traffic for this application except for internal traffic to the api server, this is an operator that submits resources to the cluster. It has the appropriate rbac permission. In this case, it seems to me the network policy doesn't need to be updated -- can you confirm?

I see the 8080 and 8081 ports are used by the application are already added in the Network Policy. If there are any other ports that are accessed by external traffic, then you would need to update the Network Policy. 

Do you have a live-build where this can be tested?

**Comment by VaishnaviHire** [comment]
/lgtm

Code changes look good

**Comment by jgarciao** [comment]
I've tested quay.io/hukhan/rhods-operator-live-catalog:1.25.0-dspo-18 and the rhods-data-science-pipelines-operator kfdef was deployed and the dspo components were deployed too

image

**Comment by diegolovison** [comment]
Using `quay.io/hukhan/rhods-operator-live-catalog:1.25.0-dspo-18`
```
data-science-pipelines-operator-controller-manager-57b6bcf4cm2k   1/1     Running   0          63m
data-science-pipelines-operator-controller-manager-57b6bcf6cgqp   1/1     Running   0          63m
data-science-pipelines-operator-controller-manager-57b6bcfcj8h6   1/1     Running   0          63m

```

**Comment by HumairAK** [comment]
Okay we should be good to merge this once https://github.com/red-hat-data-services/odh-manifests/pull/324 
is merged.

**Review by anishasthana — APPROVED** [review]
All approvals/lgtms are in place. merging.

**Pull request merged**